google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -1178,32 +1178,38 @@ public PooledSessionFuture get() {
   }
 
   class MultiplexedSessionFutureWrapper implements SessionFutureWrapper<MultiplexedSessionFuture> {
-    private SettableApiFuture<SessionInstance> multiplexedSessionSettableApiFuture;
+    private SettableApiFuture<SessionReference> multiplexedSessionSettableApiFuture;
     private ISpan span;
+    private MultiplexedSessionFuture multiplexedSessionFuture;
 
     public MultiplexedSessionFutureWrapper(
-        SettableApiFuture<SessionInstance> multiplexedSessionSettableApiFuture, ISpan span) {
+        SettableApiFuture<SessionReference> multiplexedSessionSettableApiFuture, ISpan span) {
       this.multiplexedSessionSettableApiFuture = multiplexedSessionSettableApiFuture;
       this.span = span;
     }
 
     @Override
     public MultiplexedSessionFuture get() {
-      try {
-        // Creating a new reference where the request's span state can be stored.
-        SessionImpl sessionImpl =
-            new SessionImpl(
-                sessionClient.getSpanner(), this.multiplexedSessionSettableApiFuture.get());
-        MultiplexedSession multiplexedSession = new MultiplexedSession(sessionImpl);
-        SettableFuture<MultiplexedSession> settableFuture = SettableFuture.create();
-        settableFuture.set(multiplexedSession);
-        MultiplexedSessionFuture multiplexedSessionFuture =
-            new MultiplexedSessionFuture(settableFuture, span);
+      if (multiplexedSessionFuture != null) {
```

Comment:
nit:
1. The field `multiplexedSessionFuture` should be `volatile`.
2. The lazy initialization should use a lock (the lock only needs to be taken during the actual initialization)

See this for a concrete example: https://www.baeldung.com/java-singleton-double-checked-locking#implementation

Reply:
Done